src/PruneUnusedImagesModal.jsx

```diff
@@ -49,81 +49,71 @@ function ImageOptions({ images, checked, isSystem, handleChange, name, showCheck
     );
 }
 
-class PruneUnusedImagesModal extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            deleteUserImages: this.props.userServiceAvailable !== null && this.props.userServiceAvailable,
-            deleteSystemImages: this.props.systemServiceAvailable,
-            isPruning: false,
-        };
-    }
+const PruneUnusedImagesModal = ({ close, unusedImages, onAddNotification, userServiceAvailable, systemServiceAvailable }) => {
+    const [isPruning, setPruning] = useState(false);
+    const [deleteUserImages, setDeleteUserImages] = useState(userServiceAvailable !== null && userServiceAvailable);
+    const [deleteSystemImages, setDeleteSystemImages] = useState(systemServiceAvailable);
 
-    handlePruneUnusedImages = () => {
-        this.setState({ isPruning: true });
+    const handlePruneUnusedImages = () => {
+        setPruning(true);
 
         const actions = [];
-        if (this.state.deleteUserImages) {
+        if (deleteUserImages) {
             actions.push(client.pruneUnusedImages(false));
         }
-        if (this.state.deleteSystemImages) {
+        if (deleteSystemImages) {
             actions.push(client.pruneUnusedImages(true));
         }
-        Promise.all(actions).then(this.props.close)
+        Promise.all(actions).then(close)
                 .catch(ex => {
                     const error = _("Failed to prune unused images");
-                    this.props.onAddNotification({ type: 'danger', error, errorDetail: ex.message });
-                    this.props.close();
+                    onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                    close();
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1146-20221124-132403-fa425528-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)